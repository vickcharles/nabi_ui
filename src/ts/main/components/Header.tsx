import * as React from 'react';
import { Link } from 'react-router-dom';

/** 
 * Header Stateless Component
*/
const Header = () => {
    return (
        <header>
            <Link id="nabi-logo-anchor" to="/">
                <img 
                    id="nabi-logo" 
                    className="nabi-text-center" 
                    alt="logo" 
                    src={require('../../../assets/images/logo.png')}
                />
                <p 
                    id="nabi-logo-text" 
                    className="nabi-text-center nabi-font-montserrat nabi-text-extrabold"
                >
                    Nabi Music
                </p>
            </Link>
        </header>
    );
};

export default Header;
