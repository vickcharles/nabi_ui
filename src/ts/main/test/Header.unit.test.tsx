import * as React from 'react';
import Header from '../components/Header';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';

describe('Header', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    ).find(Header).dive();
  });

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
