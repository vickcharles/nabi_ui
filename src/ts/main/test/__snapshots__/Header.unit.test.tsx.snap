// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nabi-header-container nabi-position-relative"
      >
        <DrawerMenu />
        <WithStyles(Divider)
          className="header-divider"
        />
        <Link
          id="nabi-logo-anchor"
          replace={false}
          to="/"
        >
          <img
            alt="logo"
            className="nabi-text-center"
            id="nabi-logo"
            src="logo.png"
          />
          <p
            className="nabi-text-center nabi-font-montserrat nabi-text-extrabold"
            id="nabi-logo-text"
          >
            NABI MUSIC
          </p>
        </Link>
        <div
          className="nabi-header-button"
        >
          <WithStyles(Button)
            className="nabi-responsive-button"
          >
            LOG IN
          </WithStyles(Button)>
          <Link
            replace={false}
            to="/registration"
          >
            <WithStyles(Button)
              className="nabi-responsive-button"
              color="primary"
              variant="raised"
            >
              CREATE ACCOUNT
            </WithStyles(Button)>
          </Link>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DrawerMenu />,
          <WithStyles(Divider)
            className="header-divider"
          />,
          <Link
            id="nabi-logo-anchor"
            replace={false}
            to="/"
          >
            <img
              alt="logo"
              className="nabi-text-center"
              id="nabi-logo"
              src="logo.png"
            />
            <p
              className="nabi-text-center nabi-font-montserrat nabi-text-extrabold"
              id="nabi-logo-text"
            >
              NABI MUSIC
            </p>
          </Link>,
          <div
            className="nabi-header-button"
          >
            <WithStyles(Button)
              className="nabi-responsive-button"
            >
              LOG IN
            </WithStyles(Button)>
            <Link
              replace={false}
              to="/registration"
            >
              <WithStyles(Button)
                className="nabi-responsive-button"
                color="primary"
                variant="raised"
              >
                CREATE ACCOUNT
              </WithStyles(Button)>
            </Link>
          </div>,
        ],
        "className": "nabi-header-container nabi-position-relative",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "header-divider",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="nabi-text-center"
                id="nabi-logo"
                src="logo.png"
              />,
              <p
                className="nabi-text-center nabi-font-montserrat nabi-text-extrabold"
                id="nabi-logo-text"
              >
                NABI MUSIC
              </p>,
            ],
            "id": "nabi-logo-anchor",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "nabi-text-center",
                "id": "nabi-logo",
                "src": "logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NABI MUSIC",
                "className": "nabi-text-center nabi-font-montserrat nabi-text-extrabold",
                "id": "nabi-logo-text",
              },
              "ref": null,
              "rendered": "NABI MUSIC",
              "type": "p",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                className="nabi-responsive-button"
              >
                LOG IN
              </WithStyles(Button)>,
              <Link
                replace={false}
                to="/registration"
              >
                <WithStyles(Button)
                  className="nabi-responsive-button"
                  color="primary"
                  variant="raised"
                >
                  CREATE ACCOUNT
                </WithStyles(Button)>
              </Link>,
            ],
            "className": "nabi-header-button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "LOG IN",
                "className": "nabi-responsive-button",
              },
              "ref": null,
              "rendered": "LOG IN",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  className="nabi-responsive-button"
                  color="primary"
                  variant="raised"
                >
                  CREATE ACCOUNT
                </WithStyles(Button)>,
                "replace": false,
                "to": "/registration",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "CREATE ACCOUNT",
                  "className": "nabi-responsive-button",
                  "color": "primary",
                  "variant": "raised",
                },
                "ref": null,
                "rendered": "CREATE ACCOUNT",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nabi-header-container nabi-position-relative"
        >
          <DrawerMenu />
          <WithStyles(Divider)
            className="header-divider"
          />
          <Link
            id="nabi-logo-anchor"
            replace={false}
            to="/"
          >
            <img
              alt="logo"
              className="nabi-text-center"
              id="nabi-logo"
              src="logo.png"
            />
            <p
              className="nabi-text-center nabi-font-montserrat nabi-text-extrabold"
              id="nabi-logo-text"
            >
              NABI MUSIC
            </p>
          </Link>
          <div
            className="nabi-header-button"
          >
            <WithStyles(Button)
              className="nabi-responsive-button"
            >
              LOG IN
            </WithStyles(Button)>
            <Link
              replace={false}
              to="/registration"
            >
              <WithStyles(Button)
                className="nabi-responsive-button"
                color="primary"
                variant="raised"
              >
                CREATE ACCOUNT
              </WithStyles(Button)>
            </Link>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DrawerMenu />,
            <WithStyles(Divider)
              className="header-divider"
            />,
            <Link
              id="nabi-logo-anchor"
              replace={false}
              to="/"
            >
              <img
                alt="logo"
                className="nabi-text-center"
                id="nabi-logo"
                src="logo.png"
              />
              <p
                className="nabi-text-center nabi-font-montserrat nabi-text-extrabold"
                id="nabi-logo-text"
              >
                NABI MUSIC
              </p>
            </Link>,
            <div
              className="nabi-header-button"
            >
              <WithStyles(Button)
                className="nabi-responsive-button"
              >
                LOG IN
              </WithStyles(Button)>
              <Link
                replace={false}
                to="/registration"
              >
                <WithStyles(Button)
                  className="nabi-responsive-button"
                  color="primary"
                  variant="raised"
                >
                  CREATE ACCOUNT
                </WithStyles(Button)>
              </Link>
            </div>,
          ],
          "className": "nabi-header-container nabi-position-relative",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "header-divider",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className="nabi-text-center"
                  id="nabi-logo"
                  src="logo.png"
                />,
                <p
                  className="nabi-text-center nabi-font-montserrat nabi-text-extrabold"
                  id="nabi-logo-text"
                >
                  NABI MUSIC
                </p>,
              ],
              "id": "nabi-logo-anchor",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": "nabi-text-center",
                  "id": "nabi-logo",
                  "src": "logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "NABI MUSIC",
                  "className": "nabi-text-center nabi-font-montserrat nabi-text-extrabold",
                  "id": "nabi-logo-text",
                },
                "ref": null,
                "rendered": "NABI MUSIC",
                "type": "p",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <WithStyles(Button)
                  className="nabi-responsive-button"
                >
                  LOG IN
                </WithStyles(Button)>,
                <Link
                  replace={false}
                  to="/registration"
                >
                  <WithStyles(Button)
                    className="nabi-responsive-button"
                    color="primary"
                    variant="raised"
                  >
                    CREATE ACCOUNT
                  </WithStyles(Button)>
                </Link>,
              ],
              "className": "nabi-header-button",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "LOG IN",
                  "className": "nabi-responsive-button",
                },
                "ref": null,
                "rendered": "LOG IN",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Button)
                    className="nabi-responsive-button"
                    color="primary"
                    variant="raised"
                  >
                    CREATE ACCOUNT
                  </WithStyles(Button)>,
                  "replace": false,
                  "to": "/registration",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "CREATE ACCOUNT",
                    "className": "nabi-responsive-button",
                    "color": "primary",
                    "variant": "raised",
                  },
                  "ref": null,
                  "rendered": "CREATE ACCOUNT",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
